src/main/java/com/autotune/database/helper/DBHelpers.java

```diff
@@ -690,10 +690,10 @@ public static List<DataSourceInfo> convertKruizeDataSourceToDataSourceObject(Lis
                 for (KruizeDataSource kruizeDataSource : kruizeDataSourceList) {
                     try {
                         DataSourceInfo dataSourceInfo = null;
-                        if(null != kruizeDataSource.getUrl())
-                             dataSourceInfo = new DataSourceInfo(kruizeDataSource.getName(), kruizeDataSource
+                        if (kruizeDataSource.getServiceName().isEmpty() && null != kruizeDataSource.getUrl()) {
+                            dataSourceInfo = new DataSourceInfo(kruizeDataSource.getName(), kruizeDataSource
                                     .getProvider(), new URL(kruizeDataSource.getUrl()));
-                        else{
+                        } else{
```

Comment:
What happens when serviceName is empty and URL is null?

Reply:
Datasource is not added to the database when serviceName is empty and URL is null,  and I see below pod logs -

> 2024-03-2210:28:51.751 ERROR [main][DataSourceCollection.java(187)]-Datasource servicename and url both cannot be empty.
2024-03-2210:28:51.752 INFO [main][Autotune.java(114)]-Checking available datasources:
2024-03-2210:28:52.295 WARN [main][KubernetesServicesImpl.java(425)]-Watcher not added! Only REST API access is enabled.
2024-03-2210:28:52.370 INFO [main][ExperimentManager.java(32)]-EM version: testv1
2024-03-2210:28:52.371 INFO [main][ExperimentManager.java(28)]-Initializing EM
2024-03-2210:28:52.372 INFO [main][ExperimentManager.java(71)]-Adding EM Servlets
2024-03-2210:28:52.373 INFO [main][EMExecutorService.java(30)]-Creating regular executor
2024-03-2210:28:52.373 INFO [main][EMExecutorService.java(34)]-Creating scheduled executor
2024-03-2210:28:52.374 INFO [main][EMExecutorService.java(67)]-Starting stage processors
2024-03-2210:28:52.374 INFO [main][EMExecutorService.java(67)]-Starting stage processors
Waiting for experiments ... 
2024-03-2210:29:11.059 ERROR [qtp253011924-36][ListDataSourcesAPIObject.java(25)]-No datasources found
 
 But here's the `/datasources` API response with an empty list of DataSourceInfo objects 
```
{
  "version": "v1.0"
}
```